### ISSUE TITLE
Extension crashed and then all the stored sessions gone, Don't have any backups sadly, Hope someone can help

### ISSUE THREAD

**Issue by HamedSadeghipour** [sienori/Tab-Session-Manager#1295]
Note: Since I suspect it might actually be me exhausting extension's limitations I didn't issued it as a bug report, however technically it's some kind of unexpected behaviour, so one may consider it a bug.

### Hi there. :slightly_smiling_face: I am actually not here to whine or act entitled about what I want. In fact I am very thankful for the extension you made and for the helps it gave to me. The only thing I'm trying to do is to make sure that I used the little chance there is to get my saved sessions back.
First off, I confess that I did pretty nasty things that makes the fact that the extension worked until now seem more crazy than it didn't. ***I have a very bad habit of opening the browser (MS Edge), launching so many tabs then hibernating my PC without closing any tabs, and the next day I turn the PC back on again and repeat the process***. This time I did it for about one month :sweat:. Naturally the amount of tabs became enormous and the PC became slower over time, and the ***TSM extension crashed multiple (maybe 20) times*** which I restored by turning the extension off and on again. Any time I restored the extension, my saved sessions were still there (Note that here only the extension got crashed, not the browser itself). But ***the last time TSM crashed (which was yesterday), I turned it on again and when I opened the sessions list, (much to my frustration) it was completely empty*** :cry:. Everything was gone. Naturally though after some time it started to save periodic regular auto-save sessions into the list which is expected behaviour. ***What's really odd is that even my TSM settings (e.g. Maximum number of Auto Saved tabs, etc.) are intact. So it couldn't be an extension reset*** (unless I'm wrong about how extension reset works in Edge/Chrome :thinking:). I also looked at "`%LOCALAPPDATA%\Microsoft\Edge\User Data\Default\IndexedDB`" to see if there are any remnants of the session files but it doesn't seem so because all the TSM related folders together have a size of about 3 MB which couldn't be all of my sessions (I assume it must be at least 500 MB). It's probably just a wild dream but ***Could it be that TSM temporarily moved the Tab lists somewhere else until the browser is restarted?*** Also I remember whenever the whole browser (I mean the MS Edge itself) crashed, the tab lists didn't change either, so another interesting thing is that even though the browser itself didn't crash, ***the sessions list of the TSM became empty after the extension itself crashed. Something that never happened before for me*** (although this time the extension crashed many times so maybe it's fair :slightly_frowning_face:). Actually even my current session is still alive and though I'm embarrassed to tell you how many tabs I have open now, you may want to know that currently (brace yourselves :sweat:) there are around 3200 tabs open on my browser. Yeah, disgusting, I know. And to make matters worse, because of the many hibernations I did, it seems Windows made the PageFile.sys (Virtual memory allocation file) on my C drive so large (around 50 GB) that at some point I started to see low disk space warnings so I moved a large amount of my personal data (Downloads, etc.) to let the partition breathe . So ***it could also be Windows trying to empty some space*** :worried: ***but it's weird if Windows decided to delete TSM's tab list related files specifically***. And last but not least MS Edge has two due updates waiting to be installed. So ***restraining the browser to get updated could be another reason***.

***Yeah it's messy, I know*** :cry:***, and I hope it's going to be a lesson for me to regularly back up my sessions list. I just wanted to ask you, as someone who knows how the extension works, Do you guess there's any slight chance that my tab lists are still somewhere to be found? And if it's not bothering you can you walk me through some steps to try and find my tab sessions list?***

Otherwise, I still admire your work greatly :slightly_smiling_face:. I'm serious, the first time I used your extension it was like a dream coming true and I believe all you open source developers are actual magicians who single-handedly make very useful software which are truly functional. (Unless they fall into the hands of a weirdo like me who doesn't watch for a computer's memory limitations. :neutral_face:)

Best Wishes.

**Comment by HamedSadeghipour** [comment]
Sorry for tagging/pinging you @sienori (I know you are getting a lot these days :sweat:) but since you are fortunately back would you believe whether one can recover from such a messy situation I described above (:point_up:). It could really save my life.
Thanks again for all your works. :pray:

**Comment by cocoonkid** [comment]
TLDR; please format this better, remove unnecessary information and use more lists and paragraphs.
I can understand anyone that just closes this big flippin wall of text. And I'm a speedreader.

**Comment by freemedom** [comment]
I don't think the data can be recovered. I also lost it once due to an extension reinstall...
(I also studied indexeddb for a while...

**Comment by freemedom** [comment]
The data is actually stored in indexeddb, but in your case, indexeddb is probably already overwritten.

**Comment by sienori** [comment]
I'm sorry, but I cannot provide instructions on how to restore corrupted storage. However, in some cases, the following information might be helpful. Unfortunately, I cannot offer further assistance beyond this.
https://github.com/sienori/Tab-Session-Manager/wiki/Q&A:-Sessions-data-storage-location;-How-to-migrate-sessions-data-to-another-profile
